src/main/java/teammates/ui/newcontroller/ConfirmFeedbackSessionSubmissionAction.java

```diff
@@ -0,0 +1,115 @@
+        boolean isSubmissionEmailConfirmationEmailRequested =
+                getBooleanRequestParamValue(Const.ParamsNames.SEND_SUBMISSION_EMAIL);
+        Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
+
+        EmailWrapper email = null;
+        switch (intent) {
+        case STUDENT_SUBMISSION:
+            StudentAttributes studentAttributes = getStudentOfCourseFromRequest(feedbackSession.getCourseId());
+            boolean hasStudentRespondedForSession =
+                    logic.hasGiverRespondedForSession(studentAttributes.getEmail(), feedbackSessionName, courseId);
+            if (hasStudentRespondedForSession) {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, studentAttributes.getEmail(), false, false);
+            } else {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, studentAttributes.getEmail(), false, true);
+            }
+            if (isSubmissionEmailConfirmationEmailRequested) {
+                email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForStudent(
+                            feedbackSession, studentAttributes, Instant.now());
+            }
+            break;
+        case INSTRUCTOR_SUBMISSION:
+            InstructorAttributes instructorAttributes = getInstructorOfCourseFromRequest(feedbackSession.getCourseId());
+            boolean hasInstructorRespondedForSession =
+                    logic.hasGiverRespondedForSession(instructorAttributes.getEmail(), feedbackSessionName, courseId);
+            if (hasInstructorRespondedForSession) {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, instructorAttributes.getEmail(), true, false);
+            } else {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, instructorAttributes.getEmail(), true, true);
+            }
+            if (isSubmissionEmailConfirmationEmailRequested) {
+                email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForInstructor(
+                        feedbackSession, instructorAttributes, Instant.now());
+            }
+            break;
+        default:
+            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        }
+
+        if (email != null) {
+            try {
+                emailSender.sendEmail(email);
+            } catch (EmailSendingException e) {
+                log.severe("Submission confirmation email failed to send: "
+                        + TeammatesException.toStringWithStackTrace(e));
+                return new JsonResult("Submission confirmation email failed to send",
+                        HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
I don't think you should return an error here? A failed email, while not desirable, does not mean a failed submission.